src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -68,6 +68,11 @@ public void handleRemoveEvent(CatalogRemoveEvent event) throws CatalogException
                 for (DataAccessRule r : rulesToRemove) {
                     dao.removeRule(r);
                 }
+                try {
+                    dao.storeRules();
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
```

Comment:
This should also throw an exception IMHO.

Reply:
fixed